src/runtime/crdt/crdt-entity.ts

```diff
@@ -0,0 +1,202 @@
+            op = {type: EntityOpTypes.Set, value: operation.value, actor: operation.actor, clock: operation.clock, field: singleton};
+          } else {
+            op = {type: EntityOpTypes.Clear, actor: operation.actor, clock: operation.clock, field: singleton};
+          }
+          modelOps.push(op);
+        }
+        for (const operation of singletonChanges[singleton].otherChange.operations) {
+          let op;
+          if (operation.type as SingletonOpTypes === SingletonOpTypes.Set) {
+            op = {type: EntityOpTypes.Set, value: operation.value, actor: operation.actor, clock: operation.clock, field: singleton};
+          } else {
+            op = {type: EntityOpTypes.Clear, actor: operation.actor, clock: operation.clock, field: singleton};
+          }
+          otherOps.push(op);
+        }
+      }
+      for (const collection of Object.keys(collectionChanges)) {
+        for (const operation of collectionChanges[collection].modelChange.operations) {
+          let op;
+          if (operation.type as CollectionOpTypes === CollectionOpTypes.Add) {
+            op = {type: EntityOpTypes.Add, added: operation.value, actor: operation.actor, clock: operation.clock, field: collection};
+          } else {
+            op = {type: EntityOpTypes.Remove, removed: operation.value, actor: operation.actor, clock: operation.clock, field: collection};
+          }
+          modelOps.push(op);
+        }
+        for (const operation of collectionChanges[collection].otherChange.operations) {
+          let op;
+          if (operation.type as CollectionOpTypes === CollectionOpTypes.Add) {
+            op = {type: EntityOpTypes.Add, added: operation.value, actor: operation.actor, clock: operation.clock, field: collection};
+          } else {
+            op = {type: EntityOpTypes.Remove, removed: operation.value, actor: operation.actor, clock: operation.clock, field: collection};
+          }
+          otherOps.push(op);
+        }     
+      }
+      return {modelChange: {changeType: ChangeType.Operations, operations: modelOps}, 
+              otherChange: {changeType: ChangeType.Operations, operations: otherOps}};
+    } else {
+      // need to map this.model to get the data out.
+
+      const change: EntityChange<S, C> = {changeType: ChangeType.Model, modelPostChange: this.getData()};
+      return {modelChange: change, otherChange: change};
+    }
+  }
+
+  applyOperation(op: EntityOperation<S, C>): boolean {
+    if (op.type === EntityOpTypes.Set || op.type === EntityOpTypes.Clear) {
+      if (!this.model.singletons[op.field]) {
+        if (this.model.collections[op.field as keyof C]) {
+          throw new Error(`Can't apply ${op.type === EntityOpTypes.Set ? 'Set' : 'Clear'} operation to collection field ${op.field}`);
+        }
+        throw new Error(`Invalid field: ${op.field} does not exist`);
+      }
+    } else {
+      if (!this.model.collections[op.field]) {
```

Comment:
} else if (...) {

Reply:
I actually think this is better as-is; it demonstrates more clearly the symmetry of the two clauses.